server/mem_agent/tools.py

```
@@ -0,0 +1,359 @@
+        # Ensure the file path is properly resolved
+        if not os.path.exists(file_path):
+            return f"Error: File {file_path} does not exist"
+        
+        if not os.path.isfile(file_path):
+            return f"Error: {file_path} is not a file"
+            
+        with open(file_path, "r") as f:
+            return f.read()
+    except PermissionError:
+        return f"Error: Permission denied accessing {file_path}"
+    except Exception as e:
```

Ruff (0.14.1)

Do not catch blind exception: `Exception`

(BLE001)